internal/infrastructure/logger/gorm_log.go

```diff
@@ -2,92 +2,73 @@ package logger
 
 import (
 	"context"
-	"fmt"
-	"os"
+	"errors"
 	"time"
 
-	"github.com/podengo-project/idmsvc-backend/internal/config"
-	"github.com/rs/zerolog"
+	"golang.org/x/exp/slog"
+	"gorm.io/gorm"
 	"gorm.io/gorm/logger"
 )
 
-type gormZerolog struct {
-	logger zerolog.Logger
+type gormLogger struct {
+	slogger              *slog.Logger
+	ignoreRecordNotFound bool
 }
 
-func NewGormLog(cfg *config.Config) logger.Interface {
-	var o *os.File
-	if cfg.Logging.Console {
-		o = os.Stdout
-	} else {
-		o = os.Stderr
-	}
-
-	lvl, err := zerolog.ParseLevel(cfg.Logging.Level)
-	if err != nil {
-		panic(err)
-	}
-	l := zerolog.New(o).
-		Level(lvl).
-		With().
-		Timestamp().
-		Logger()
-
-	return &gormZerolog{
-		logger: l,
-	}
+func (l *gormLogger) LogMode(level logger.LogLevel) logger.Interface {
+	return l
 }
 
-func (l *gormZerolog) LogMode(level logger.LogLevel) logger.Interface {
-	switch level {
-	case logger.Silent:
-		{
-			l.logger.Level(zerolog.NoLevel)
-		}
-	case logger.Info:
-		{
-			l.logger.Level(zerolog.InfoLevel)
-		}
-	case logger.Warn:
-		{
-			l.logger.Level(zerolog.WarnLevel)
-		}
-	case logger.Error:
-		{
-			l.logger.Level(zerolog.ErrorLevel)
-		}
-	}
-	return l
+func (l *gormLogger) Info(ctx context.Context, msg string, args ...interface{}) {
+	l.slogger.InfoCtx(ctx, msg, args)
```

Comment:
- In latest version, `*Ctx` variants like `InfoCtx` are deprecated in favor of `InfoContext`, `WarnContext`, etc
- gorm uses format-style log messages. You either need to use `Infof()` function or manually `fmt.Sprintf` the message and arguments.
- slog includes the location of a logger call. You have to jump through extra hoops to make slog report the correct location for wrappers. https://github.com/golang/exp/blob/master/slog/example_wrap_test.go has an example how to implement a log wrapper.

Reply:
I've implemented it

Reply:
The change to print out the location is getting unexpected result; a new ticket is addressing this; I got the below:

```raw
{"time":"2023-11-30T17:25:07.731265779+01:00","level":"ERROR","msg":"failed to initialize database, got error failed to connect to `host=localhost user=idmsvc database=idmsvc`: server error (FATAL: password authentication failed for user \"idmsvc\" (SQLSTATE 28P01))"}
```

Reply:
HMS-3196 filed